src/Model/AutoLoginKey.php

```diff
@@ -78,9 +91,31 @@ public static function get_by_key( $key ) {
 			return null;
 		}
 
+		if ( (bool) $row['one_time'] ) {
+			self::maybe_delete_one_time_key( $key );
```

Comment:
I feel that the call to maybe_delete_one_time_key should be moved to the method handle_auto_login in the Autologin class after the following line:
`'do_action( 'wp_login', $user->user_login, $user );' `
It seems like the key should only be deleted **after** successfully logging the user in.
Consider the unlikely case where the login key and user id passed in do not match. In that case the login would fail but the key would still be deleted.